client/js/verification-portal/components/time.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react'
+import injectSheet from 'react-jss'
+import { connect } from 'react-redux'
+import { change } from 'redux-form'
+
+class Time extends React.Component {
+	render() {
+		const { classes, input, placeholder, changeFieldValue } = this.props
+		let i = 0
+		const hr = [
+			'12:30',
+			'1:00',
+			'1:30',
+			'2:00',
+			'2:30',
+			'3:00',
+			'3:30',
+			'4:00',
+			'4:30',
+			'5:00',
+			'5:30',
+			'6:00',
+			'6:30',
+			'7:00',
+			'7:30',
+			'8:00',
+			'8:30',
+			'9:00',
+			'9:30',
+			'10:00',
+			'10:30',
+			'11:00',
+			'11:30',
+			'12:00',
+		]
+		const timeSpin = (current, name, up) => {
+			if (!current)
+				return name.split('.')[1] == 'open'
+					? changeFieldValue(name, '9:00 am')
+					: changeFieldValue(name, '5:00 pm')
+			current = current.split(' ')
+			i = hr.indexOf(current[0])
+			if (up == 'up') {
+				if (!hr[i - 1]) {
+					i = hr.length
+					current[1] == 'am' ? (current[1] = 'pm') : (current[1] = 'am')
+				}
+				current[0] = hr[i - 1]
+			} else {
+				if (!hr[i + 1]) {
+					i = -1
+					current[1] == 'am' ? (current[1] = 'pm') : (current[1] = 'am')
+				}
+				current[0] = hr[i + 1]
+			}
+			current = current.join(' ')
+			changeFieldValue(name, current)
+		}
```

Comment:
@bhavyaab Nice job getting this to work! 👏

We're going to need to clean this function up though so that it is more readable and maintainable. 😄 I'm having a hard time figuring out what is actually going on here. A couple of suggestions:
- This function can be pulled out of the class (we should avoid nested functions as much as possible)
- The naming of variables is extremely important in knowing what the value represents and helps a lot with readability. I would encourage you to think of more descriptive (and probably longer) variable names.
- If possible, nested `if` statements should be avoided. By refactoring the code to remove the nested ifs, it's likely it will become more readable.

Let me know if you have any questions or would like any help in cleaning this up, I would be more than happy to help!